# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1546909648 28800
#      Mon Jan 07 17:07:28 2019 -0800
# Branch stable
# Node ID 6b703927b891dd4c58566903b6120e74157b11e2
# Parent  0ae92ab9dddc11bcae6a1e2a12be4aaffe4da356
test: use workaround to allow pkg.tst to pass in 'make distcheck' (bug #53578)

* pkg.tst: Ensure temporary pkg prefix directory is writeable before attempting
to uninstall packages.  This is a temporary hack that uses recursive chmod to
work around the real issue to allow the tests to pass.

diff --git a/test/pkg/pkg.tst b/test/pkg/pkg.tst
--- a/test/pkg/pkg.tst
+++ b/test/pkg/pkg.tst
@@ -56,25 +56,27 @@
 %!function silent_pkg_install (varargin)
 %!  evalc (["pkg install", sprintf(" %s", varargin{:})]);
 %!endfunction
 
 ## Action install/uninstall
 %!testif HAVE_Z
 %! for i = 1:numel (mfile_pkg_name)
 %!   silent_pkg_install (mfile_pkg_tgz{i});
+%!   system (["chmod -Rf u+w '" prefix "'"]);   ## FIXME: Work around bug #53578
 %!   pkg ("uninstall", mfile_pkg_name{i});
 %! endfor
 %!
 %!error pkg ("install", "nonexistent.zip")
 
 # -local
 %!testif HAVE_Z
 %! for i = 1:numel (mfile_pkg_name)
 %!   silent_pkg_install ("-local", mfile_pkg_tgz{i});
+%!   system (["chmod -Rf u+w '" prefix "'"]);   ## FIXME: Work around bug #53578
 %!   pkg ("uninstall", mfile_pkg_name{i});
 %! endfor
 
 # -forge (need check for options?)
 ## FIXME: Need test
 # We do not test this yet ... fails if no internet connection
 # use dataframe which is an mfile only package
 #%!test
@@ -85,24 +87,25 @@
 ## FIXME: Need test
 
 # -verbose
 ## FIXME: Need test
 
 ## Action load/unload (within install/uninstall)
 %!testif HAVE_Z
 %! for i = 1:numel (mfile_pkg_name)
-%!  name = mfile_pkg_name{i};
-%!  silent_pkg_install ("-local", mfile_pkg_tgz{i});
-%!  unwind_protect
-%!    pkg ("load", name);
-%!    pkg ("unload", name);
-%!  unwind_protect_cleanup
-%!    pkg ("uninstall", name);
-%!  end_unwind_protect
+%!   name = mfile_pkg_name{i};
+%!   silent_pkg_install ("-local", mfile_pkg_tgz{i});
+%!   unwind_protect
+%!     pkg ("load", name);
+%!     pkg ("unload", name);
+%!   unwind_protect_cleanup
+%!     system (["chmod -Rf u+w '" prefix "'"]); ## FIXME: Work around bug #53578
+%!     pkg ("uninstall", name);
+%!   end_unwind_protect
 %! endfor
 %!
 %!error <package foobar is not installed> pkg ("load", "foobar");
 
 # -nodeps
 ## FIXME: Need test
 
 # -verbose
@@ -117,16 +120,17 @@
 #%! oct_forge_pkgs = pkg ("list", "-forge");
 
 ## Action describe
 %!testif HAVE_Z
 %! silent_pkg_install ("-local", mfile_pkg_tgz{1});
 %! [desc, flag] = pkg ("describe", mfile_pkg_name{1});
 %! ## FIXME: this only tests that the describe command runs,
 %! ##        not that the output is in anyway correct.
+%! system (["chmod -Rf u+w '" prefix "'"]);     ## FIXME: Work around bug #53578
 %! pkg ("uninstall", mfile_pkg_name{1});
 
 # -verbose
 ## FIXME: Need test
 
 ## Action prefix
 %!test
 %! pfx_old = pkg ("prefix");
